appl/cmd/gunzip.b

```diff
@@ -34,7 +34,7 @@ init(nil: ref Draw->Context, argv: list of string)
 	if (bufio == nil)
 		fatal(sys->sprint("cannot load %s: %r", Bufio->PATH));
 	str = load String String->PATH;
-	if (bufio == nil)
+	if (str == nil)
 		fatal(sys->sprint("cannot load %s: %r", String->PATH));
```

Comment:
Please forgive my ignorance... I'm just starting in Inferno.

This looks incorrect: `sys->sprint("cannot load %s: %r", String->PATH)`. The `%s` is the string for `String->PATH`. But what is the argument to `%r`? It seems to be missing. (Assuming Limbo `sprintf` resembles C's `sprintf`).

Reply:
No problem. It's a good question, and it's not obvious.
The `%r` format conversion token outputs the most recent system error message. See section 2.5 of [A Descent into Limbo](https://www.vitanuova.com/inferno/papers/descent.html) or section 12.2 of [The Limbo Programming Language](https://www.vitanuova.com/inferno/papers/limbo.html).

Reply:
See sys-print(2)